src/driver-steelseries.c

```diff
@@ -65,6 +65,74 @@
 #define STEELSERIES_BUTTON_KBD			0x51
 #define STEELSERIES_BUTTON_CONSUMER		0x61
 
+struct steelseries_point {
+	struct list link;
+
+	struct ratbag_color color;
+	uint8_t pos;
+};
+
+struct steelseries_cycle {
+	uint8_t led_id; 			/* led id */
+	uint16_t duration;			/* cycle duration */
+	bool repeat;				/* if the cycle restarts automatically */
+	uint8_t trigger_buttons;	/* trigger button combination */
+	struct list points;			/* colors in the cycle */
+};
+
+static void
+create_cycle(struct steelseries_cycle *cycle)
+{
+	cycle->led_id = 0x00;
+	cycle->duration = 5000;
+	cycle->repeat = true;
+	cycle->trigger_buttons = 0x00;
+
+	list_init(&cycle->points);
+}
+
+static void
+construct_cycle_buffer(struct steelseries_cycle *cycle, uint8_t *buf)
+{
+	struct steelseries_point *point;
+	uint16_t duration, min_duration;
+	uint8_t point_count = 0;
+
+	buf[0] = STEELSERIES_ID_LED;
+	buf[2] = cycle->led_id;
+
+	if (!cycle->repeat)
+		buf[19] = 0x01;
+
+	buf[23] = cycle->trigger_buttons;
+
+	list_for_each(point, &cycle->points, link) {
+		if(point_count == 0){
+			buf[28] = point->color.red;
+			buf[29] = point->color.green;
+			buf[30] = point->color.blue;
+		}
```

Comment:
I'd appreciate a comment here to make clear that this isn't a missing a `continue;` statement here and that we really want to write the first point at two locations.